array_api_compat/common/_helpers.py

```diff
@@ -156,8 +174,17 @@ def device(x: "Array", /) -> "Device":
     out: device
         a ``device`` object (see the "Device Support" section of the array API specification).
     """
-    if _is_numpy_array(x):
+    if is_numpy_array(x):
         return "cpu"
+    if is_jax_array(x):
+        # JAX has .device() as a method, but it is being deprecated so that it
+        # can become a property, in accordance with the standard. In order for
+        # this function to not break when JAX makes the flip, we check for
+        # both here.
```

Comment:
Does this logic seem OK?

Reply:
This looks OK, but it will not work with the `to_device` function as currently defined in `jax.experimental.array_api`. That expects to be passed the bound method object – it's a hack, and one of the reasons that this is still considered experimental.

Reply:
The other problem here is that, in general, JAX arrays can live on multiple devices (in which case `arr.device()` will error). It's not clear from the Array API design how multi-device arrays are meant to be supported.

Reply:
It looks like both forms work

```py
>>> import jax.experimental.array_api as xp
>>> x = xp.asarray([1, 2, 3])
>>> x.to_device(x.device)
Array([1, 2, 3], dtype=int32)
>>> x.to_device(x.device())
Array([1, 2, 3], dtype=int32)
```